Feature/187184253

goku673 opened the pull request:
when is clicked on the email it should display a button "cerrar sesion", this button should delete the token of the current userand redirect to properties page.

goku673 committed:
When is clicked on the email it should display a button

when is clicked on the email it should display a button "cerrar sesion", this button should delete the token of the current userand redirect to properties page.

douglascamposh commented on src/components/TopMenu.js:
```diff
@@ -21,15 +27,36 @@ const TopMenu = () => {
+   useEffect( ()=>{
+      if(tokenData !== null) {
+        const dataToken = JSON.parse(atob(tokenData.split('.')[1]));
+        console.log(dataToken?.sub);
```
remove console log

goku673 replied:
okey

douglascamposh commented on src/components/TopMenu.js:
```diff
@@ -44,20 +71,27 @@ const TopMenu = () => {
+
             ))}
           </ul>
           {isLoggedIn ? (
```
is really needed the `isLoggedIn` flag maybe just we can put `tokenDataEmail ?`. what do you think?

goku673 replied:
I'm gonna check it

douglascamposh commented on src/components/TopMenu.js:
```diff
@@ -44,20 +71,27 @@ const TopMenu = () => {
+              <button onClick={handleButtonEmailActivate}>
+                <span className="mr-2">{tokenDataEmail}</span>
+              </button>
+              {butttonEmailActivate && (
```
is really needed a flag to display the dropdown maybe you could add a sub menu and add the `cerrar sesion` item.

goku673 committed:
when is clicked on the email it should display a button

when is clicked on the email it should display a button "cerrar sesion", this button should delete the token of the current userand redirect to properties page.

goku673 committed:
when is clicked on the email it should display a button

Whe is clicked on the email it should display a button "cerrar seion", this button is inside a Dropdown, this button should delete the token of the current user and redirect to properties page.

goku673 committed:
remove unnecessary line of code

goku673 committed:
new dropDownMenu component

Drop-down menu where the logout button is displayed.

douglascamposh commented on src/components/TopMenu.js:
```diff
@@ -6,14 +6,15 @@ import NavItem from "./NavBar/NavLink";
   const [menuIcon, setMenuIcon] = useState(false);
   const [tokenDataEmail, setTokenDataEmail] = useState(null);
-  const [butttonEmailActivate, setButtonEmailActivate] = useState(false);
+  //const [butttonEmailActivate, setButtonEmailActivate] = useState(true);
```
remove commented code

douglascamposh commented on src/components/DropdownMenu.js:
```diff
@@ -0,0 +1,26 @@
+
+import { MdExitToApp, MdChevronRight } from "react-icons/md";
+
+const DropdownMenu = ({ handleLogout, tokenDataEmail }) => {
```
looks good, but maybe we could create DropdownItem instead of  to have the button and the option fixed.
`
<li>
                        <button onClick={handleLogout} className="flex items-center w-full p-2 text-left text-sm gap-2 border border-primary hover:text-primary">
                            <div className="rounded-full p-1 bg-slate-400 flex items-center">
                                <MdExitToApp className="text-white" size={24} />
                            </div>
                            <span className="flex-grow" style={{ whiteSpace: 'nowrap' }}>Cerrar Sesión</span>
                            <MdChevronRight className="ml-3" size={24} />
                        </button>
                    </li>
` this could be a component and we can send as parameters the handlelogout and the text or use a for each and recieve an array with the textb and the event

goku673 committed:
optimization in DropdownMenu component

douglas-campos-coderoad-com committed:
refactor dropdown
